### ISSUE TITLE
deploy failed - Removing persistent bind mounts

### ISSUE THREAD

**Issue by drnic** [cloudfoundry/bosh-lite#253]
Using latest bosh-lite (and using old v388) I saw this during a `bosh deploy` (of concourse):

```
  Started compiling packages
  Started compiling packages > bosh_deployment_resource/86c173a276e78a3f8b7458ed275bd53596b313ac
  Started compiling packages > bosh_io_stemcell_resource/27625ff4c39126a3286d4657abcde5955f1e6f94
  Started compiling packages > bosh_io_release_resource/963ee9cd703fbf35302278c789d3311799d9617a
     Done compiling packages > bosh_io_stemcell_resource/27625ff4c39126a3286d4657abcde5955f1e6f94 (00:00:10)
  Started compiling packages > vagrant_cloud_resource/158236da0390451b2ad4fba64ac7fffe04b0b328
     Done compiling packages > bosh_io_release_resource/963ee9cd703fbf35302278c789d3311799d9617a (00:00:19)
  Started compiling packages > tracker_resource/7f6425c98c8bd70f58b114389d1b1713fa8acb4a
     Done compiling packages > bosh_deployment_resource/86c173a276e78a3f8b7458ed275bd53596b313ac (00:00:19)
  Started compiling packages > time_resource/62013ba59dabc6f0f3ca66a1dc18d4b91b4631aa
     Done compiling packages > vagrant_cloud_resource/158236da0390451b2ad4fba64ac7fffe04b0b328 (00:00:10)
  Started compiling packages > semver_resource/1707ec45b97a8b9044ba24ca661f6d4fb89dcc16
     Done compiling packages > time_resource/62013ba59dabc6f0f3ca66a1dc18d4b91b4631aa (00:00:06)
  Started compiling packages > s3_resource/3ed7b3d8f115b979c83752c5d9019b248ed3dbb8
     Done compiling packages > semver_resource/1707ec45b97a8b9044ba24ca661f6d4fb89dcc16 (00:00:07)
  Started compiling packages > github_release_resource/be1cd5002e9c6a89f674692af322637133b94ac6
   Failed compiling packages > tracker_resource/7f6425c98c8bd70f58b114389d1b1713fa8acb4a: Deleting vm '4242d372-f08e-426a-4b48-0a873aff0594': Removing persistent bind mounts: remove /var/vcap/store/cpi/persistent_bind_mounts_dir/4242d372-f08e-426a-4b48-0a873aff0594: device or resource busy (00:00:09)
     Done compiling packages > github_release_resource/be1cd5002e9c6a89f674692af322637133b94ac6 (00:00:06)
     Done compiling packages > s3_resource/3ed7b3d8f115b979c83752c5d9019b248ed3dbb8 (00:00:08)

Error 100: Deleting vm '4242d372-f08e-426a-4b48-0a873aff0594': Removing persistent bind mounts: remove /var/vcap/store/cpi/persistent_bind_mounts_dir/4242d372-f08e-426a-4b48-0a873aff0594: device or resource busy
```

**Comment by drnic** [comment]
Redeploying results in this error:

```
  Started preparing deployment > Binding existing deployment. Failed: Timed out sending `get_state' to 7409aeea-2830-4ebd-abdc-d5e0d1959432 after 45 seconds (00:02:15)
```

This requires deleting the deployment and redeploying.

**Comment by bosh-ci-push-pull** [comment]
I've seen this in the concourse pipeline I have not shared for
bosh-warden-cpi. Interesting that this issue was not popping up before.
I'll try to add some retry logic to see if that helps.

**Comment by drnic** [comment]
Still getting this for every fresh deploy of concourse on bosh-lite.

**Comment by drnic** [comment]
Current workaround is to guess now many times it will fail and repeat the delete-deploy cycle until it completes :/

https://github.com/drnic/concourse-bosh-lite/blob/master/setup.sh#L24-L26

**Comment by cppforlife** [comment]
Almost there to make this change: https://main.bosh-ci.cf-app.com/pipelines/warden

**Comment by cppforlife** [comment]
I believe this should be fixed in 9000.35.0+ with bosh-warden-cpi release 21+ (https://github.com/cppforlife/bosh-warden-cpi/commit/e1423b4327b82919bd6e57eba0433fc7b06dae85)
